text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+
+A quick proposal of what this might look like:
+
+- `target_info` metric generation is updated to exclude any keys which are contained in `descriptive_attributes_keys` of an entity.
+- For each entity which has non-empty descriptive_attributes_keys, generate an info metric: `<entity_type>_entity_info` (naming TBD), which has all identifying and descriptive keys. This should play nicely with the planned improvements to [info-typed metrics](https://github.com/prometheus/proposals/blob/main/proposals/2024-04-10-native-support-for-info-metrics-metadata.md#goals).
+
+### Should entities have a domain?
+
+Is it worth having a `domain` in addition to type for entity?  We could force each entity to exist in one domain and leverage domain generically in resource management.  Entity Detectors would be responsible for an entire domain, selecting only ONE to apply a resource. Domains could be layered, e.g. a Cloud-specific domain may layer on top of a Kubernetes domain, where "GKE cluster entity" identifies *which* kubernetes cluster a kuberntes infra entity is part of.  This layer would be done naively, via automatic join of participating entities or explicit relationships derived from GKE specific hooks.
+
+It's unclear if this is needed initially, and we believe this could be layered in later.
+
+### Should resources have only one associated entity?
+
+Given the problems leading to the Entities working group, and the needs of existing Resource users today, we think it is infeasible and unscalable to limit resource to only one entity.  This would place restrictions on modeling Entities that would require OpenTelemetry to be the sole source of entity definitions and hurt building an open and extensible ecosystem.  Additionally it would need careful definition of solutions for the following problems/rubrics:
+
+- New entities added by extension should not break existing code
+- Collector augmentation / enrichment (resource, e.g.) - Should be extensible and not hard-coded. We need a general algorithm not specific rulesets.
+
+### What identity should entities use (LID, UUID / GUID, or other)?
+
+One of the largest questions in the first entities' OTEP was how to identify an entity.  This was an attempt to unify the need for Navigational attributes with the notion that only identifying attributes of Entity would show up in Resource going forward. This restriction is no longer necessary in this proposal and we should reconsider how to model identity for an Entity.  
+
+This can be done in follow up design / OTEPs.
+
+### What happens if existing Resource translation in the collector remove resource attributes an Entity relies on?
```

Comment:
The Collector's behavior that addresses this problem is explained in a previous document [here](https://docs.google.com/document/d/1VUdBRInLEhO_0ABAoiLEssB1CQO_IcD5zDnaMEha42w/edit#heading=h.7jq82arq6q09). Any reason not to go with the suggested approach? I think it solves this problem.

Reply:
With the change from storing key-value pairs on Entity to just key references, this no longer works in the backwards compatible case (old collector, new data).